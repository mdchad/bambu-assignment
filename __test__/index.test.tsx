import React from 'react';
import {render, cleanup, fireEvent} from "@testing-library/react";
import "@testing-library/jest-dom";
import Home from '../pages/index';

afterEach(cleanup);

it("matches snapshot", () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
});

it('allows input', () => {
    const { getByTestId } = render(<Home />)
    let item = 'New York'
    const searchInputElement = getByTestId('search-input')
    searchInputElement.value = item;
    fireEvent.change(searchInputElement)
    expect(searchInputElement.value).toBe('New York')
})
