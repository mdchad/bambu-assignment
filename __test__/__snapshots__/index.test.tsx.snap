// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="m-auto antialiased font-sans font-serif font-mono text-center bg-gray-900 min-h-screen flex flex-col items-center text-white text-2xl"
    >
      <div
        class="mt-10 flex flex-col"
      >
        <h1
          class="text-3xl mb-2"
        >
          Job Search
        </h1>
        <form
          class="ml-4 flex flex-row"
          id="input-styling-address"
        >
          <div>
            <span
              class="algolia-places-nostyle"
              style="display: inline-block;"
            >
              <input
                aria-autocomplete="both"
                aria-expanded="false"
                aria-label="Enter a city"
                aria-owns="algolia-places-nostyle-listbox-0"
                autocomplete="off"
                class="flex-3 bg-gray-200 hover:bg-white hover:border-gray-300 focus:outline-none focus:bg-white focus:shadow-outline focus:border-gray-300 appearance-none border border-transparent rounded w-full py-2 px-4 text-gray-700 leading-tight ap-nostyle-input"
                data-testid="search-input"
                dir="auto"
                placeholder="Enter a city"
                role="combobox"
                spellcheck="false"
                style="vertical-align: top;"
                type="text"
              />
              <pre
                aria-hidden="true"
                style="visibility: hidden; white-space: pre; font-family: -webkit-small-control; word-spacing: normal; letter-spacing: normal; text-indent: 0; text-rendering: auto; text-transform: none;"
              />
              <span
                class="ap-nostyle-dropdown-menu"
                id="algolia-places-nostyle-listbox-0"
                role="listbox"
                style="top: 100%; left: 0px; z-index: 100; display: none;"
              >
                <div
                  class="ap-nostyle-dataset-places"
                />
              </span>
              <button
                aria-label="clear"
                class="ap-nostyle-input-icon ap-nostyle-icon-clear"
                style="display: none;"
                type="button"
              >
                <svg
                  height="12"
                  viewBox="0 0 12 12"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M.566 1.698L0 1.13 1.132 0l.565.566L6 4.868 10.302.566 10.868 0 12 1.132l-.566.565L7.132 6l4.302 4.3.566.568L10.868 12l-.565-.566L6 7.132l-4.3 4.302L1.13 12 0 10.868l.566-.565L4.868 6 .566 1.698z"
                  />
                </svg>
                

              </button>
              <button
                aria-label="focus"
                class="ap-nostyle-input-icon ap-nostyle-icon-pin"
                type="button"
              >
                <svg
                  height="20"
                  viewBox="0 0 14 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7 0C3.13 0 0 3.13 0 7c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5C5.62 9.5 4.5 8.38 4.5 7S5.62 4.5 7 4.5 9.5 5.62 9.5 7 8.38 9.5 7 9.5z"
                  />
                </svg>
                

              </button>
            </span>
          </div>
          <button
            class="flex-1 mx-4 bg-indigo-600 hover:bg-indigo-700 text-white font-bold py-2 px-4 rounded"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
      <div
        class="my-32 mx-24"
      >
        <div
          class="max-w-sm w-full lg:max-w-full lg:flex"
        >
          <div
            class="w-full border-r border-b border-l border-gray-400 lg:border-l-0 lg:border-t lg:border-gray-400 bg-white rounded-b lg:rounded-b-none lg:rounded-r p-4 flex flex-col justify-between leading-normal"
          >
            <h1
              class="text-gray-900 text-left font-bold text-xl mb-2"
            >
              No Jobs Found
            </h1>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
